commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -0,0 +1,90 @@
+package fr.free.nrw.commons.upload;
+
+import android.media.ExifInterface;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+public class GPSExtractor {
+
+    private String filePath;
+
+    public GPSExtractor(String filePath){
+        this.filePath = filePath;
+    }
+
+    //Extract GPS coords of image
+    public String getCoords() {
+
+        ExifInterface exif;
+        String latitude = "";
+        String longitude = "";
+        String latitude_ref = "";
+        String longitude_ref = "";
+        String decimalCoords = "";
+
+        try {
+            exif = new ExifInterface(filePath);
+            latitude = exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE);
+            latitude_ref = exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE_REF);
+            longitude = exif.getAttribute(ExifInterface.TAG_GPS_LONGITUDE);
+            longitude_ref = exif.getAttribute(ExifInterface.TAG_GPS_LONGITUDE_REF);
+
+            Log.d("Image", "Latitude: " + latitude + " " + latitude_ref);
+            Log.d("Image", "Longitude: " + longitude + " " + longitude_ref);
+
+            decimalCoords = getDecimalCoords(latitude, latitude_ref, longitude, longitude_ref);
+
+        } catch (IOException e) {
+            Log.w("Image", e);
+        }
+        return decimalCoords;
+    }
+
+    //Converts format of coords into decimal coords as required by API for next step
+    private String getDecimalCoords(String latitude, String latitude_ref, String longitude, String longitude_ref) {
```

Comment:
It's a little strange to use snake_case here and camelCase everywhere else.  I think the convention is camelCase.